ReactiveCocoa/DelegateProxy.swift

```diff
@@ -53,57 +54,64 @@ extension DelegateProxy {
 	internal static func proxy<P: DelegateProxy<Delegate>>(
 		for instance: NSObject,
 		setter: Selector,
-		getter: Selector,
-		_ key: StaticString = #function
+		getter: Selector
 	) -> P {
-		return _proxy(for: instance, setter: setter, getter: getter, key) as! P
+		return _proxy(for: instance, setter: setter, getter: getter) as! P
 	}
 
 	private static func _proxy(
 		for instance: NSObject,
 		setter: Selector,
-		getter: Selector,
-		_ key: StaticString = #function
+		getter: Selector
 	) -> AnyObject {
-		let key = AssociationKey<DelegateProxy<Delegate>?>(key)
-
 		return instance.synchronized {
-			if let proxy = instance.associations.value(forKey: key) {
+			let proxyMap = instance.associations.value(forKey: proxyMapKey)
+
+			if let proxy = proxyMap.first(where: { $0.0 == setter })?.1 {
 				return proxy
 			}
 
+			let invokeOriginalSetter: @convention(c) (NSObject, Selector, AnyObject?) -> Void = { object, selector, delegate in
+				typealias Setter = @convention(c) (NSObject, Selector, AnyObject?) -> Void
+				let impl = class_getMethodImplementation(object.objcClass, selector)
+				unsafeBitCast(impl, to: Setter.self)(object, selector, delegate)
+			}
+
 			let newSetterImpl: @convention(block) (NSObject, AnyObject?) -> Void = { object, delegate in
-				let proxy = object.associations.value(forKey: key)!
-				proxy.forwardee = (delegate as! Delegate?)
+				let proxyMap = object.associations.value(forKey: proxyMapKey)
+
+				if let proxy = proxyMap.first(where: { $0.0 == setter })?.1 as! DelegateProxy<Delegate>? {
```

Comment:
Okay, I understand what's happening now.

It seems like we could instead create an `AssociationKey` from `setter` since it's a `Selector`.

If for some reason that doesn't work, we should be able to use `[Selector: AnyObject]` since `Selector` is `Hashable`.